text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,329 @@
+
+With continuously compounded interest, we _could_ update the interest in `on_finalize` as we did before, but it would need to be updated every single block. Instead we wait until a user tries to use the account (to deposit or withdraw funds), and then calculate the account's current value "just in time".
+
+To facilitate this implementation, we represent the state of the account not only as a balance, but as a balance, paired with the time when that balance was last updated.
+
+```rust, ignore
+#[derive(Encode, Decode, Default)]
+pub struct ContinuousAccountData<BlockNumber> {
+	/// The balance of the account after last manual adjustment
+	principle: I32F32,
+	/// The time (block height) at which the balance was last adjusted
+	deposit_date: BlockNumber,
+}
+```
+
+You can see we've chosen substrate-fixed's `I32F32` as our balance type this time. While we don't intend to handle negative balances, there is currently a limitation in the transcendental functions that requires using signed types.
+
+With the struct to represent the account's state defined, we can initialize the storage value.
+
+```rust, ignore
+decl_storage! {
+	trait Store for Module<T: Trait> as Example {
+		// --snip--
+
+		/// Balance for the continuously compounded account
+		ContinuousAccount get(fn balance_compound): ContinuousAccountData<T::BlockNumber>;
+	}
+}
+```
+
+As before, there are two relevant extrinsics, `deposit_continuous` and `withdraw_continuous`. They are nearly identical so we'll only show one.
+
+```rust, ignore
+fn deposit_continuous(origin, val_to_add: u64) -> DispatchResult {
+	ensure_signed(origin)?;
+
+	let current_block = system::Module::<T>::block_number();
+	let old_value = Self::value_of_continuous_account(&current_block);
+
+	// Update storage for compounding account
+	ContinuousAccount::<T>::put(
+		ContinuousAccountData {
+			principle: old_value + I32F32::from_num(val_to_add),
+			deposit_date: current_block,
+		}
+	);
+
+	// Emit event
+	Self::deposit_event(Event::DepositedContinuous(val_to_add));
+	Ok(())
+}
+```
+
+This function itself isn't too insightful. It does the same basic things as the discrete variant: lookup the old value, and the deposit, update storage, emit event. The one interesting part is that it calls a helper function to get the account's previous value. This helper function calculates the value of the account considering all the interest that has accrued since the last time the account was touched. Let's take a closer look.
```

Comment:
```suggestion
This function itself isn't too insightful. It does the same basic things as the discrete variant: look up the old value and the deposit, update storage, and emit an event. The one interesting part is that it calls a helper function to get the account's previous value. This helper function calculates the value of the account considering all the interest that has accrued since the last time the account was touched. Let's take a closer look.
```